peer_controller.go

```diff
@@ -433,6 +433,18 @@ func (p *peerController) markHtlcComplete(ctx context.Context, key circuitKey,
 		return
 	}
 
+	// If we couldn't look up an outgoing peer for the HTLC, either:
+	// 1. The outgoing channel never existed (since this is not validated on intercept)
+	// 2. The outgoing channel is pending close at time of resolution (edge case)
+	//
+	// Log and skip this HTLC for now to avoid adding empty values to the DB.
+	if resolution.outgoingPeer == nil {
+		log.Debugf("HTLC: %v has no known outgoing peer, not storing resolution",
```

Comment:
Are there always two log lines now for this case?

Reply:
Ah yes there are, cutting this one.